### DISCUSSION TITLE
Repeater mode

### DISCUSSION THREAD

**Discussion by f4dvk** [alanbjohnston/CubeSatSim#343] [General]
Hi,

When I activate repeater mode, the BF is not retransmitted.
How does this mode work?

Thanks,
Stephane.

**Comment by alanbjohnston** [comment]
Hi Stephane,

You can try out the experimental repeater mode if you have a Beta v1.3 or later CubeSatSim (blue boards) and you have added the Command and Control hardware described here: https://github.com/alanbjohnston/CubeSatSim/wiki/Command-and-Control

By BF I presume you mean your Baofeng FM HT radio.  That radio should work.

First, make sure you are transmitting on the right frequency.  On your Pi Zero, type this command:

`CubeSatSim/config`

And look for the two frequencies listed in the sim.cfg file.  The first one (default 434.9 MHz) is the CubeSatSim transmit frequency.  The second one (default 435 MHz) is the receive frequency.  Also make sure there are no PL Tones set (the two numbers after the frequency should both be 0).

Next, check to see if your transmitter breaks the squelch on the CubeSatSim FM transceiver.  You can do this by typing this command and watching to see if the Squelch goes to zero when you key up an FM transmission:

`CubeSatSim/config -Q`

If you see the squelch being broken, then you can test the Repeater mode by this command:

`CubeSatSim/config -e`

Now when you break the squelch with your HT, you should hear an output on the CubeSatSim transmit frequency.  The level can be a bit low, so you might need to raise the volume to hear it.  You can press DTMF keys on your transmitter as they are sent at a high level.  Sometimes the recorder crashes and then you will only hear a carrier retransmitted.  A reboot should fix this if it happens.

If you don't hear anything, take a look at the transmitter log with this command to see if you have any errors:

`CubeSatSim/log -r`

My code is based on https://zr6aic.blogspot.com/2016/11/creating-2m-fm-repeater-with-raspberry.html so you can try out his examples as well, although they use the RTL-SDR as a receiver.

Good luck!

Alan

**Comment by f4dvk** [comment]
Hi Alan,

It works after a reboot, but there are a lot of crashes.

Thanks,
Stephane.

**Comment by alanbjohnston** [comment]
Hi Stephane,

I made some code changes that should eliminate the audio crashes.  Let me know if you still have failures.  An update should do the trick:

`CubeSatSim/update`

Alan

**Comment by njsat** [comment]
Thank you for the update.

CubSatNJ
Thank you,

Rogers E. Campbell, IV
RECMedia, LLC
Cell: (973) 202-9384
Pager: (201) 237-0581
